tensorflow/core/kernels/mkl/mkl_quantized_conv_ops_test.cc

```diff
@@ -436,43 +459,50 @@ TEST_F(QuantizedConv2DTest, Small32Bit) {
-                                   output_quantized);
+  if (!NativeFormatEnabled()) {
+    const Tensor& output_mkl_metadata = *GetOutput(3);
 
-  test::ExpectTensorEqual<qint32>(expected, output_quantized);
+    ConvMklToTF conv_comp;
+    Tensor output_quantized;
+    conv_comp.ConvertMklToTF<qint32>(DT_QINT32, output, output_mkl_metadata,
+                                     output_quantized);
+    test::ExpectTensorEqual<qint32>(expected, output_quantized);
+  } else {
+    test::ExpectTensorEqual<qint32>(expected, output);
+  }
 }
 
 // Output -> qint32
 TEST_F(QuantizedConv2DTest, Small32BitWithPadding) {
   const int stride = 1;
-  TF_ASSERT_OK(NodeDefBuilder("quantized_conv_op", "_MklQuantizedConv2D")
-                   .Input(FakeInput(DT_QUINT8))  // Input
-                   .Input(FakeInput(DT_QINT8))   // Filter
-                   .Input(FakeInput(DT_FLOAT))   // Min input
-                   .Input(FakeInput(DT_FLOAT))   // Max input
-                   .Input(FakeInput(DT_FLOAT))   // Min filter
-                   .Input(FakeInput(DT_FLOAT))   // Max filter
-                   //  MKL metadata tensors //
-                   .Input(FakeInput(DT_UINT8))
-                   .Input(FakeInput(DT_UINT8))
-                   .Input(FakeInput(DT_UINT8))
-                   .Input(FakeInput(DT_UINT8))
-                   .Input(FakeInput(DT_UINT8))
-                   .Input(FakeInput(DT_UINT8))
-                   ///////////////////////////
-                   .Attr("Tinput", DataTypeToEnum<quint8>::v())
-                   .Attr("Tfilter", DataTypeToEnum<qint8>::v())
-                   .Attr("T", DataTypeToEnum<quint8>::v())
-                   .Attr("out_type", DataTypeToEnum<qint32>::v())
-                   .Attr("strides", {1, stride, stride, 1})
-                   .Attr("padding", "SAME")
-                   .Attr("padding_list", {0, 0, 1, 1, 1, 1, 0, 0})
-                   .Attr("_kernel", "QuantizedMklOp")
-                   .Finalize(node_def()));
+  NodeDefBuilder builder =
+      NodeDefBuilder("quantized_conv_op", NativeFormatEnabled()
+                                              ? "_MklNativeQuantizedConv2D"
+                                              : "_MklQuantizedConv2D")
+          .Input(FakeInput(DT_QUINT8))  // Input
+          .Input(FakeInput(DT_QINT8))   // Filter
+          .Input(FakeInput(DT_FLOAT))   // Min input
+          .Input(FakeInput(DT_FLOAT))   // Max input
+          .Input(FakeInput(DT_FLOAT))   // Min filter
+          .Input(FakeInput(DT_FLOAT))   // Max filter
+          .Attr("Tinput", DataTypeToEnum<quint8>::v())
+          .Attr("Tfilter", DataTypeToEnum<qint8>::v())
+          .Attr("T", DataTypeToEnum<quint8>::v())
+          .Attr("out_type", DataTypeToEnum<qint32>::v())
+          .Attr("strides", {1, stride, stride, 1})
+          .Attr("padding", "SAME")
+          .Attr("padding_list", {0, 0, 1, 1, 1, 1, 0, 0})
+          .Attr("_kernel", "QuantizedMklOp");
```

Comment:
Should `T` and `_kernel` be in `if (!NativeFormatEnabled())` instead?

Reply:
For `T` yes. For `_kernel` no, per the MKL rewrite algorithm, label is required if the op name changes (i.e. adding _MklNative).